python/pyxir/contrib/target/components/DPUCAHX8H/u50lv_op_support.py

```diff
@@ -0,0 +1,235 @@
+def mean_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided Mean operator
+    on the DPUCAHX8H-u50lv target"""
+    axes = X.attrs["axes"]
+    keepdims = X.attrs["keepdims"]
+    return len(axes) == 2 and keepdims
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u50lv", "LeakyReLU")
+def leaky_relu_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided LeakyRelu operator
+    on the DPUCAHX8H-u50lv target"""
+    assert len(bXs) == 1
+    bX = bXs[0]
+    alpha = X.attrs["alpha"]
+    # LeakyRelu not supported after depthwise conv2d
+    return math.isclose(alpha, 0.1, rel_tol=1e-5) and bX.type[0] in [
+        "Convolution",
+        "Conv2DTranspose",
+        "BatchNorm",
+        "BiasAdd",
+        "Scale",
+    ]
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u50lv", "pReLU")
+def prelu_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided pRelu operator
+    on the DPUCAHX8H-u50lv target"""
+    # Only LeakyReLU: alpha == 0.1 supported
+    assert len(bXs) == 1
+    bX = bXs[0]
+    alpha = X.attrs["alpha"]
+    # LeakyRelu not supported after depthwise conv2d
+    return math.isclose(alpha, 0.1, rel_tol=1e-5) and bX.type[0] in [
+        "Convolution",
+        "Conv2DTranspose",
+        "BatchNorm",
+        "BiasAdd",
+        "Scale",
+    ]
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u50lv", "ReLU")
+def relu_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided ReLU operator
+    on the DPUCAHX8H-u50lv target"""
+    assert len(bXs) == 1
+    bX = bXs[0]
+    return bX.type[0] in set(
+        ["Convolution", "Conv2DTranspose", "Eltwise", "BatchNorm", "BiasAdd", "Scale"]
+    )
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u50lv", "ReLU6")
+def relu6_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided ReLU6 operator
+    on the DPUCAHX8H-u50lv target"""
+    assert len(bXs) == 1
+    bX = bXs[0]
+    return bX.type[0] in set(
+        ["Convolution", "Conv2DTranspose", "Eltwise", "BatchNorm", "Scale"]
+    )
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u50lv", "Scale")
+def scale_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided Scale operator
+    on the DPUCAHX8H-u50lv target"""
+    return is_scale_supported(X, bXs, tXs, channel_parallel=16, bank_depth=2048)
+
+
+@pyxir.register_op_support_check("DPUCAHX8H-u50lv", "Upsampling2D")
+def upsampling_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """Check whether we can execute the provided Upsampling2D operator
+    on the DPUCAHX8H-u50lv target"""
+    # TODO
+    method = X.attrs["method"]
+    return method == "nearest_neighbor"
```

Comment:
We should check for:  `ceil(scale / bank_num) * scale * ceil(input_channel / channel_parallel) <= bank_depth`